controllers/helpers.go

```diff
@@ -126,6 +129,50 @@ func validateLabelSelector(selector labels.Selector) error {
 	return nil
 }
 
+// transformCloudSpecToHostsSpec from a cloud spec disruption, get all ip ranges of services provided and transform them into a list of hosts spec
+func transformCloudSpecToHostsSpec(log *zap.SugaredLogger, cloudManager *cloudservice.CloudProviderManager, cloudSpec *v1beta1.NetworkDisruptionCloudSpec) []v1beta1.NetworkDisruptionHostSpec {
+	hosts := []v1beta1.NetworkDisruptionHostSpec{}
+	clouds := map[cloudtypes.CloudProviderName][]string{
+		cloudtypes.CloudProviderAWS:     cloudSpec.AWS,
+		cloudtypes.CloudProviderDatadog: cloudSpec.Datadog,
+		cloudtypes.CloudProviderGCP:     cloudSpec.GCP,
+	}
+
+	for cloudName, serviceList := range clouds {
+		var ips []string
+
+		var err error
+
+		if len(serviceList) == 0 {
+			ips, err = cloudManager.GetServiceIPRanges(cloudName, "")
+			if err != nil {
+				log.Errorf("could not retrieve the ip range of all services of %s", cloudName)
+				continue
+			}
+		}
+
+		for _, service := range serviceList {
+			tmpIps, err := cloudManager.GetServiceIPRanges(cloudName, service)
+			if err != nil {
+				log.Errorf("could not retrieve the ip range of %s for the service %s", cloudName, service)
+				continue
+			} else {
+				ips = append(ips, tmpIps...)
+			}
+		}
+
+		for _, ip := range ips {
+			hosts = append(hosts, v1beta1.NetworkDisruptionHostSpec{
+				Host:     ip,
+				Protocol: "tcp",
+				Flow:     "egress",
+			})
+		}
+	}
+
+	return hosts
+}
+
```

Comment:
Same problem as the comment above:

Added this in helpers.go instead of the specific GenerateArgs() method of the NetworkDisruptionSpec [here](https://github.com/DataDog/chaos-controller/blob/d934f1534a70126846d2f81493147e3777bce6c8/api/v1beta1/network_disruption.go#L136) due to a problem with the package deepcopy-gen. It's preventing us from passing the cloudProviderManager to those methods because we can't generate the DeepCopyInto method on interfaces.
In my opinion, we should put this in GenerateArgs() but I'm not sure the time wasted trying to find a workaround is worth it.